diff --git a/Release/CMakeLists.txt b/Release/CMakeLists.txt
index 18b0cb3f..35c08bef 100644
--- a/Release/CMakeLists.txt
+++ b/Release/CMakeLists.txt
@@ -122,6 +122,9 @@ if(ANDROID)
   set(ANDROID_LIBS atomic dl)
 elseif(UNIX) # This includes OSX
 elseif(WIN32)
+  if(NOT CPPREST_EXCLUDE_COMPRESSION)
+    add_definitions(-DCPPREST_HTTP_COMPRESSION)
+  endif()
   add_definitions(-DUNICODE -D_UNICODE -DWIN32 -D_SCL_SECURE_NO_WARNINGS)
   if(CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
     add_definitions(-D_WIN32_WINNT=0x0A00)
