diff --git a/Release/src/http/client/http_client_winhttp.cpp b/Release/src/http/client/http_client_winhttp.cpp
index 249bfcd5..d4e66aed 100644
--- a/Release/src/http/client/http_client_winhttp.cpp
+++ b/Release/src/http/client/http_client_winhttp.cpp
@@ -1108,6 +1108,7 @@ protected:
         DWORD ignoredCertificateValidationSteps = 0;
         if (client_config().validate_certificates())
         {
+#if 0
             // if we are validating certificates, also turn on revocation checking
             DWORD dwEnableSSLRevocationOpt = WINHTTP_ENABLE_SSL_REVOCATION;
             if (!WinHttpSetOption(winhttp_context->m_request_handle,
@@ -1119,6 +1120,7 @@ protected:
                 request->report_error(errorCode, build_error_msg(errorCode, "Error enabling SSL revocation check"));
                 return;
             }
+#endif
 
             // check if the user has overridden the desired Common Name with the host header
             const auto hostHeader = headers.find(_XPLATSTR("Host"));
