diff --git a/Release/src/http/client/http_client_winhttp.cpp b/Release/src/http/client/http_client_winhttp.cpp
index 61a13abf..a8591154 100644
--- a/Release/src/http/client/http_client_winhttp.cpp
+++ b/Release/src/http/client/http_client_winhttp.cpp
@@ -1757,6 +1757,11 @@ private:
                 }
             }
         }
+        else
+        {
+            // streambuf::getpos returns EOF on error, which means that the stream is probably not seekable.
+            return false;
+        }
         p_request_context->m_compression_state = winhttp_request_context::compression_state();
 
         //  If we got ERROR_WINHTTP_RESEND_REQUEST, the response header is not available,
